### PR TITLE
Constrain BankUS.routing_number's first two digits

### PR DESCRIPTION
**Pull request by professor**
Mea Culpa! I went to use `BankUS.routing_number` with our application and had 100s of test failures. Today, I learned that the first two numbers need to be within a certain range. 

Before submitting this PR, I have tested the code in my application and can confirm that there were no errors in our gazillions of tests.

----

I would like feedback on the implementation, e.g. sample.

### CODE REVIEW

File: lib/ffaker/bank_us.rb [hunk 1]
```diff
@@ -10,10 +10,14 @@ def account_number(min_digits: 9, max_digits: 17)
     end
 
     def routing_number
-      partial_routing_number = FFaker.numerify('########')
-      ninth_digit = generate_ninth_digit(partial_routing_number)
+      first_two_digit_range = ((0..12).to_a + (21..32).to_a + (61..72).to_a + [80])
+      first_two_digits = format('%02d', first_two_digit_range.sample)
```
Comment (marocchino) [inline lib/ffaker/bank_us.rb:14]:
```suggestion
      first_two_digits = rand([*"00".."12",*"21".."32",*"61".."72","80"])
```

File: lib/ffaker/bank_us.rb [hunk 2]
```diff
@@ -10,10 +10,14 @@ def account_number(min_digits: 9, max_digits: 17)
...
 
-      "#{partial_routing_number}#{ninth_digit}"
+      partial_routing_number = FFaker.numerify('######')
+
+      ninth_digit = generate_ninth_digit(first_two_digits + partial_routing_number)
```
Comment (marocchino) [inline lib/ffaker/bank_us.rb:18]:
```suggestion
      partial_routing_number = FFaker.numerify("#{first_two_digits}######")

      ninth_digit = generate_ninth_digit(partial_routing_number)
```

File: lib/ffaker/bank_us.rb [hunk 3]
```diff
@@ -10,10 +10,14 @@ def account_number(min_digits: 9, max_digits: 17)
...
+
+      "#{first_two_digits}#{partial_routing_number}#{ninth_digit}"
```
Comment (marocchino) [inline lib/ffaker/bank_us.rb:20]:
```suggestion
      partial_routing_number = FFaker.numerify("#{first_two_digits}######")
      ninth_digit = generate_ninth_digit(partial_routing_number)

      "#{partial_routing_number}#{ninth_digit}"
```

File: test/test_bank_us.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@ class TestBankUS < Test::Unit::TestCase
 
   assert_methods_are_deterministic(
     FFaker::BankUS,
-    :account_number, :routing_number
+    :account_number
```
Comment (marocchino) [inline test/test_bank_us.rb:10]: All exposed methods needs to be deterministic.

File: lib/ffaker/bank_us.rb (hunk 1 above)
Comment (professor) [reply lib/ffaker/bank_us.rb:14]: Brilliant. Oddly google search was a fail on this one.

File: lib/ffaker/bank_us.rb [hunk 5]
```diff
@@ -10,7 +10,9 @@ def account_number(min_digits: 9, max_digits: 17)
     end
 
     def routing_number
-      partial_routing_number = FFaker.numerify('########')
+      first_two_digits = fetch_sample([*'00'..'12', *'21'..'32', *'61'..'72', '80'])
```
Comment (marocchino) [inline lib/ffaker/bank_us.rb:13]: maybe) 
We can declare this as constants and use it in the test. Reducing the logic used in tests will improve readability. 
```rb
VALID_ROUTING_FIRST_TWO_DIGITS=[*'00'..'12', *'21'..'32', *'61'..'72', '80']
# ....

fetch_sample(VALID_ROUTING_FIRST_TWO_DIGITS)
```

**Review by marocchino — APPROVED** [review]
LGTM

File: lib/ffaker/bank_us.rb (hunk 5 above)
Comment (professor) [reply lib/ffaker/bank_us.rb]: Nice. Fixed.

File: test/test_bank_us.rb (hunk 4 above)
Comment (professor) [reply test/test_bank_us.rb]: Thanks for showing me how to make the test deterministic.

**Pull request merged**